src/worker.cpp

```diff
@@ -1020,8 +1063,17 @@ std::tuple<bool, std::string>
             logging::logMessage(ex.what());
         }
 
-        // TODO: before returning, we should prime the inventry for FRUs which
+        // Prime the inventry for FRUs which
         // are not present/processing had some error.
+        types::ObjectMap l_objectInterfaceMap;
+        primeInventory(i_vpdFilePath, l_objectInterfaceMap);
```

Comment:
Calling it here can crash the service in some situations. Please fix that.

Reply:
Updated .

Reply:
I have mistakenly updated the comment here. 
Currently updated primeInventory API with success or failure status now 
and added log  to print if prime inventory fails .